### DISCUSSION TITLE
Is there a way to make SvgPath from text?

### DISCUSSION THREAD

**Discussion by KillyMXI** [vvvv/SVG#944] [Q&A]
Somewhat related to #908 - I can turn my strings into [GraphicsPath](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.drawing2d.graphicspath), so I hoped for a way to construct SvgPath from that.

Actual problem: PDF files produced with Svg.Skia are way too big, single page with some vector graphics and some short texts is about 2MB which is unacceptable for my client's use case (And it can get worse considering produced documents will be in different languages including Chinese).
If I get down to available Skia PDF options - they are only about raster elements, nothing of help there.
Relevant issue: https://github.com/mono/SkiaSharp/issues/848
Workaround with a custom Skia build to expose Font shrinking options seems to complicated - I have another option that will cause less trouble - skip SVG and Skia completely and use a different library to generate PDF from my data, albeit it has different downsides.
Workaround with manual font preparation - on a fence whether I should explore it further - not happy about it either.

So, naturally, I started to look how can I skip dealing with fonts all together.
In my case, produced documents don't have to retain machine-readable/searchable text, so paths should suffice.

Is there anything useful for my problem already?
Does it look convincing to provide such feature? (And where it should belong?)

If no shortcuts are available, then I should probably go ahead and build my own converter from GraphicsPath to SvgPath.
Are there different path representations that might be more convenient than GraphicsPath for this task?

**Comment by H1Gdev** [comment]
This library does not provide the ability to generate `SvgPath` from `GraphicsPath`.
If you want to generate `GraphicsPath` from `SvgPath`, you can use [AddToPath](https://github.com/svg-net/SVG/blob/f95291e6a54365503c25c1162f1e64c5d23ffa41/Source/Paths/SvgPathSegment.Drawing.cs#L36).

(FYI)
There is [codes](https://github.com/svg-net/SVG/blob/f95291e6a54365503c25c1162f1e64c5d23ffa41/Source/Paths/SvgQuadraticCurveSegment.Drawing.cs#L40-L45) that is analyzing `GraphicsPath`. It may be helpful.

**Reply by KillyMXI** [reply]
Yeah, after asking this question I went ahead and built the converter using GraphicsPath.
PathPointType seems to be poorly documented but, after figuring that out, the rest was pretty easy.

Since I'm using Svg.Skia and conversion to PDF was the whole reason for this, my main concern now is the handling of SDC dependency across the packages. I'm stuck with Svg.Skia 0.5.10 for now (the last version built with SDC), observing how things will unfold down the line.
